drivers/SmartThings/deepsmart/src/utils.lua

```diff
@@ -230,6 +230,20 @@ function utils.deep_table_eq(tbl1, tbl2)
   return false
 end
 
+-- check if valid ipv4
+function utils.is_valid_ipv4(ip) 
```

Comment:
This function already exists in our `net_utils` module: `require "net_utils"`
```lua
--- Validate dotted-decimal IPv4 address string
---@param ipv4_addr string address to validate
---@return true if valid IPv4 address, false otherwise
function net_utils.validate_ipv4_string(ipv4_addr)
```